Warn users about peer dependencies

joaomoreno opened the pull request:
It took me a while to find out that [expressjs/express-session](https://github.com/expressjs/express-session), [expressjs/cookie-parser](https://github.com/expressjs/body-parser) and [expressjs/body-parser](https://github.com/expressjs/body-parser) are required by this middleware. It's very hard to detect this, there's no error message at all.

msotclas commented:
Hi __@joaomoreno__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Joao Moreno). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla2.msopentech.com.
    </p>

TTYL, MSOTBOT;

joaomoreno commented:
- Added those 3 dependencies as [peer dependencies](http://blog.nodejs.org/2013/02/07/peer-dependencies/)
- Remove deprecated `express` functionality
- Updated sample code in `README.md`
- Updated samples

brandwe commented:
I took this but unfortunately this fell too far down the stack to be integrated automatically. I'll add you to the thank list if thats ok?
